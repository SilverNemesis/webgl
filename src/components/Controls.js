import React from 'react';

const Controls = ({ show, children }) => {
  if (!show) {
    return null;
  }
  return (
    <div id="overlay">
      {children}
    </div>
  );
}

export default Controls;
