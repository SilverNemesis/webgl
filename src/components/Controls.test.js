import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import Controls from './Controls'

const renderer = new ShallowRenderer();

it('renders correctly', () => {
    const showControls = true;
    const onClickPrevious = () => { };
    const onClickNext = () => { };
    const onChange = () => { };
    const options = [
        {
            name: 'Select Test',
            type: 'select',
            options: ['gold', 'silver', 'platinum']
        },
        {
            name: 'Bool Test',
            id: 'boolTest',
            type: 'bool'
        },
        {
            name: 'Int test',
            id: 'intTest',
            type: 'int',
            min: 0,
            max: 32
        },
        {
            name: 'Float Test',
            id: 'floatTest',
            type: 'float',
            min: 0.0,
            max: 1.0
        },
        {
            name: 'Function Test',
            type: 'function',
            function: null
        }
    ];
    renderer.render(<Controls show={showControls} onClickPrevious={onClickPrevious} onClickNext={onClickNext} onChange={onChange} options={options} />)
    const tree = renderer.getRenderOutput()
    expect(tree).toMatchSnapshot()
});
