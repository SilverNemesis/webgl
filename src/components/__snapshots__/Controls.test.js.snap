// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="overlay"
>
  <div
    className="none"
  >
    <span
      className="left"
      onClick={[Function]}
    >
      ❮ PREV
    </span>
    <span
      className="right"
      onClick={[Function]}
    >
      NEXT ❯
    </span>
  </div>
  <div>
    <SelectControl
      control={
        Object {
          "name": "Select Test",
          "options": Array [
            "gold",
            "silver",
            "platinum",
          ],
          "type": "select",
        }
      }
      onChange={[Function]}
    />
    <BoolControl
      control={
        Object {
          "id": "boolTest",
          "name": "Bool Test",
          "type": "bool",
        }
      }
      onChange={[Function]}
    />
    <IntControl
      control={
        Object {
          "id": "intTest",
          "max": 32,
          "min": 0,
          "name": "Int test",
          "type": "int",
        }
      }
      onChange={[Function]}
    />
    <FloatControl
      control={
        Object {
          "id": "floatTest",
          "max": 1,
          "min": 0,
          "name": "Float Test",
          "type": "float",
        }
      }
      onChange={[Function]}
    />
    <FunctionControl
      control={
        Object {
          "function": null,
          "name": "Function Test",
          "type": "function",
        }
      }
    />
  </div>
</div>
`;
